src/server/streamableHttp.test.ts

```diff
@@ -1214,6 +1214,148 @@ describe("StreamableHTTPServerTransport", () => {
+        params: {
+          clientInfo: { name: "test-client", version: "1.0" },
+          protocolVersion: "2025-03-26"
+        },
+        id: "init-1",
+      };
+
+      const initReq = createMockRequest({
+        method: "POST",
+        headers: {
+          "content-type": "application/json",
+          "accept": "application/json, text/event-stream",
+        },
+        body: JSON.stringify(initMessage),
+      });
+
+      mockResponse = createMockResponse();
+      await jsonResponseTransport.handleRequest(initReq, mockResponse);
+      mockResponse = createMockResponse(); // Reset for tests
+    });
+
+    it("should return JSON response for a single request", async () => {
+      const requestMessage: JSONRPCMessage = {
+        jsonrpc: "2.0",
+        method: "test",
+        params: {},
+        id: "test-req-id",
+      };
+
+      const req = createMockRequest({
+        method: "POST",
+        headers: {
+          "content-type": "application/json",
+          "accept": "application/json, text/event-stream",
+          "mcp-session-id": jsonResponseTransport.sessionId,
+        },
+        body: JSON.stringify(requestMessage),
+      });
+
+      // Mock immediate response
+      jsonResponseTransport.onmessage = (message) => {
+        if ('method' in message && 'id' in message) {
+          const responseMessage: JSONRPCMessage = {
+            jsonrpc: "2.0",
+            result: { value: "test-result" },
+            id: message.id,
+          };
+          setTimeout(() => {
+            void jsonResponseTransport.send(responseMessage);
+          }, 10);
+        }
+      };
+
+      await jsonResponseTransport.handleRequest(req, mockResponse);
+
+      // Wait for all promises to resolve
+      await new Promise(resolve => setTimeout(resolve, 50));
+
+      // Should respond with application/json header
+      expect(mockResponse.writeHead).toHaveBeenCalledWith(
+        200,
+        expect.objectContaining({
+          "Content-Type": "application/json",
+        })
+      );
+
+      // Should return the response as JSON
+      const expectedResponse = {
+        jsonrpc: "2.0",
+        result: { value: "test-result" },
+        id: "test-req-id",
+      };
+
+      expect(mockResponse.end).toHaveBeenCalledWith(JSON.stringify(expectedResponse));
+    });
+
+    it("should return JSON response for batch requests", async () => {
+      const batchMessages: JSONRPCMessage[] = [
+        { jsonrpc: "2.0", method: "test1", params: {}, id: "req1" },
+        { jsonrpc: "2.0", method: "test2", params: {}, id: "req2" },
+      ];
+
+      const req = createMockRequest({
+        method: "POST",
+        headers: {
+          "content-type": "application/json",
+          "accept": "application/json, text/event-stream",
+          "mcp-session-id": jsonResponseTransport.sessionId,
+        },
+        body: JSON.stringify(batchMessages),
+      });
+
+      // Mock sequential responses - send them in specific order
```

Comment:
There is no order guarantee for JSONRPC batch responses:
https://www.jsonrpc.org/specification#batch
> The Response objects being returned from a batch call MAY be returned in any order within the Array.
Order should not be assumed in tests nor client implementations